src/lib/shim/shadow_sem.c

```diff
@@ -0,0 +1,158 @@
+#include "lib/shim/shadow_sem.h"
+
+#include "lib/logger/logger.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <limits.h>
+#include <linux/futex.h>
+#include <semaphore.h>
+#include <stdalign.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+// These could just be at global scope, but calling this at the specific points
+// where we use  the futex is a bit less fragile.
+static inline void _value_is_valid_futex(const shadow_sem_t* s) {
```

Comment:
Are there checks you are missing in this predicate? Right now the function is a noop because the `static_assert()` calls are compiled away once the constant-expression condition is checked.

Reply:
Yes, this was intended as a noop and was just a function so that I could reference the assertions in both of the places they mattered, without duplicating them.

I changed this to be an accessor instead; hopefully that's clearer.